libraries/timeslider/src/components/TimeSlider/designer.ts

```diff
@@ -0,0 +1,223 @@
+import {
+    applyComponentModelDesignerSettings,
+    ApplyDesignerSettingsCallback,
+    ComponentModelDesignerSettings,
+    DesignerSettings,
+    getComponentModelDesignerSettings,
+    getComponentModelDesignerSettingsSchema,
+    GetDesignerSettingsCallback,
+    GetDesignerSettingsSchemaCallback,
+    Setting,
+    SettingsSchema,
+} from "@vertigis/web/designer";
+
+import TimeSliderModel from "./TimeSliderModel";
+
+export type TimeSliderLayout = "auto" | "compact" | "wide";
+export type TimeSliderMode =
```

Comment:
Rather than duplicating the Esri definitions, you can instead reference the originals:
```
export type TimeSliderMode = EsriTimeSlider["mode"];
export type TimeIntervalUnit = EsriTimeInterval["unit"];
// And so on.
```